datadog-opentelemetry/tests/integration_tests/metrics.rs

```diff
@@ -0,0 +1,232 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use datadog_opentelemetry::configuration::Config;
+use datadog_opentelemetry::metrics_exporter::OtlpProtocol;
+use datadog_opentelemetry::metrics_reader::create_meter_provider_with_protocol;
+use libdd_trace_utils::test_utils::datadog_test_agent::DatadogTestAgent;
+use opentelemetry::global;
+use opentelemetry::metrics::{Counter, Histogram, UpDownCounter};
+use opentelemetry::KeyValue;
+
+use crate::integration_tests::make_test_agent;
+
+async fn setup_test_agent(session_name: &'static str) -> DatadogTestAgent {
+    make_test_agent(session_name).await
+}
+
+fn create_all_metric_types(meter: &opentelemetry::metrics::Meter) {
+    let counter: Counter<u64> = meter.u64_counter("test.counter").build();
+    counter.add(10, &[KeyValue::new("key1", "value1")]);
+
+    let updown_counter: UpDownCounter<i64> = meter.i64_up_down_counter("test.updown").build();
+    updown_counter.add(5, &[KeyValue::new("key2", "value2")]);
+    updown_counter.add(-2, &[KeyValue::new("key2", "value2")]);
+
+    let histogram: Histogram<f64> = meter.f64_histogram("test.histogram").build();
+    histogram.record(1.5, &[KeyValue::new("key3", "value3")]);
+    histogram.record(2.3, &[KeyValue::new("key3", "value3")]);
+    histogram.record(3.7, &[KeyValue::new("key3", "value3")]);
+
+    let _observable_counter = meter
+        .u64_observable_counter("test.observable_counter")
+        .with_callback(|result| {
+            result.observe(20, &[KeyValue::new("key4", "value4")]);
+        })
+        .build();
+
+    let _observable_gauge = meter
+        .f64_observable_gauge("test.observable_gauge")
+        .with_callback(|result| {
+            result.observe(42.0, &[KeyValue::new("key5", "value5")]);
+        })
+        .build();
+
+    let _observable_updown = meter
+        .i64_observable_up_down_counter("test.observable_updown")
+        .with_callback(|result| {
+            result.observe(15, &[KeyValue::new("key6", "value6")]);
+        })
+        .build();
+}
+
+#[tokio::test]
+async fn test_metrics_export_grpc() {
+    const SESSION_NAME: &str = "opentelemetry_api/test_metrics_grpc";
+
+    let test_agent = setup_test_agent(SESSION_NAME).await;
```

Comment:
The test agent version/client we use doesn't have the otlp endppints yet.
The support has been merged in libdatadog but we need to bump it here (we won't need a new release, just update to a git hash probably)